src/kemal/middleware/filters.cr

```diff
@@ -35,18 +47,25 @@ module Kemal::Middleware
     def initialize(&@block : HTTP::Server::Context -> _)
     end
   end
+
 end
 
 def add_filters
-  Kemal.config.add_handler Kemal::Middleware::Filter.new
+  unless filter = Kemal.config.handlers.any? { |handler| handler.is_a? Kemal::Middleware::Filter }
+    filter = Kemal::Middleware::Filter.new
```

Comment:
Talking about the `unless` scope here.

`Kemal.config.add_handler Kemal::Middleware::Filter.new` is OK!